### PR TITLE
Foederati 

### PR DESCRIPTION
**Pull request by IhateTrains**
- code cleaning
- Mastimani holy sites (currently same as hellenic_pagan)
- Foederati work
- on_actions cleaning
- integrated some vanilla characters to fix history errors
- history fixes
- other minor fixes

### CODE REVIEW

File: WTWSMS/decisions/WtWSMS_diploaction_decisions.txt [hunk 1]
```diff
@@ -96,4 +96,84 @@ targetted_decisions = {
...
+		only_independent = yes
+		ai_check_interval = 12
+
+		potential = {
+			OR = {
+				has_landed_title = e_byzantium
+				has_landed_title = e_wre
+				has_landed_title = e_roman_empire
+			}
+		}
```
Comment (loup99) [inline WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: I don't think the foederati should be limited to the Roman empire itself, given that Soissons had such relations as well. Personally I think this could model that: 
```
AND = {
	primary_title = { higher_tier_than = DUKE }
	has_romance_culture_trigger = yes
}

```

File: WTWSMS/decisions/WtWSMS_diploaction_decisions.txt [hunk 2]
```diff
@@ -96,4 +96,84 @@ targetted_decisions = {
...
+		
+		effect = {
+			make_tributary = {
+				who = FROM
+				tributary_type = foederati
+			}
+			custom_tooltip = { text = chinese_grace_become_tributary_effect_TT }
+			FROM = {
+				sound_effect = china_grace_gain
+			}
```
Comment (loup99) [inline WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: Why does this connect to China?

File: WTWSMS/decisions/WtWSMS_diploaction_decisions.txt [hunk 3]
```diff
@@ -96,4 +96,84 @@ targetted_decisions = {
...
+		
+		ai_will_do = {
+			factor = 0
+			modifier = {
+				factor = 0.1 # slow down
+			}
+			modifier = { realm_levy_diff = { who = FROM value = 2000 } factor = 1.5 }
+			modifier = { realm_levy_diff = { who = FROM value = 5000 } factor = 3 }
+		}
+	}
```
Comment (loup99) [inline WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: I think having the Roman Nostalgia trait should make a bordering realm more likely to start negotiations.

File: WTWSMS/decisions/WtWSMS_diploaction_decisions.txt [hunk 4]
```diff
@@ -96,4 +96,84 @@ targetted_decisions = {
...
+					}
+				}
+				NOT = {
+					is_tributary = yes
+				}
+				NOT = { tier = emperor }
+				custom_tooltip = {
+					text = become_foederati_barbarian_cooldown_tt
+					NOT = { has_character_modifier = become_foederati_timer }
+				}
```
Comment (loup99) [inline WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: The cooldown of 30 years seems like a far too harsh one. This was a very fluid process, which could change from year to year. If we are to abstract slightly, ten years (normal truce) seems more reasonable.

File: WTWSMS/decisions/WtWSMS_diploaction_decisions.txt (hunk 2 above)
Comment (IhateTrains) [reply WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: I forgot to replace it with the name of a dedicated tooltip.

File: WTWSMS/decisions/WtWSMS_diploaction_decisions.txt (hunk 4 above)
Comment (IhateTrains) [reply WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: Done.

File: WTWSMS/decisions/WtWSMS_diploaction_decisions.txt (hunk 2 above)
Comment (IhateTrains) [reply WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: Done.

File: WTWSMS/decisions/WtWSMS_diploaction_decisions.txt (hunk 3 above)
Comment (IhateTrains) [reply WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: Done.

File: WTWSMS/decisions/WtWSMS_diploaction_decisions.txt (hunk 1 above)
Comment (IhateTrains) [reply WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: Done.

File: WTWSMS/decisions/WtWSMS_diploaction_decisions.txt [hunk 5]
```diff
@@ -96,4 +96,79 @@ targetted_decisions = {
...
+		}
+		
+		effect = {
+			make_tributary = {
+				who = FROM
+				tributary_type = foederati
+			}
+			custom_tooltip = { text = become_foederati_barbarian_effect_TT }
+			FROM = {
+				sound_effect = china_grace_gain
```
Comment (loup99) [inline WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: You kept this sound effect, was it intentional? Keep in mind that we can't use sound elements from DLCs (in this case Jade Dragon) without requiring them.

File: WTWSMS/map/geographical_region.txt [hunk 6]
```diff
@@ -629,7 +629,7 @@ custom_volcanic = {
 	duchies = {
 		d_alexandria d_damietta d_cairo
 		d_mesopotamia d_osrhoene
-		d_mauria d_tingitana d_caesariensis d_ghana
```
Comment (loup99) [inline WTWSMS/map/geographical_region.txt]: Why did you remove d_mauria?

Comment (IhateTrains) [reply WTWSMS/map/geographical_region.txt]: duplicate (it's inside the active_volcanic region already)

File: WTWSMS/decisions/WtWSMS_diploaction_decisions.txt (hunk 5 above)
Comment (IhateTrains) [reply WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: It should be replaced with other sound (would be fine if it was in base game).

Comment (IhateTrains) [reply WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: Oh grace_gain is in base vanilla.

Comment (IhateTrains) [reply WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: Done, replaced.

Comment (loup99) [reply WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: OK, if it, fine.

File: WTWSMS/map/geographical_region.txt (hunk 6 above)
Comment (loup99) [reply WTWSMS/map/geographical_region.txt]: Ah, I had not seen that.

File: WTWSMS/common/landed_titles/landed_titles.txt [hunk 7]
```diff
@@ -6115,7 +6118,7 @@ e_byzantium = {
 				color2={ 255 255 255 }
 				
 				aramaic = "Kharka"
-				khalka = "Kharka"
+				#khalka = "Kharka" #what culture should it be, khalan?
```
Comment (loup99) [inline WTWSMS/common/landed_titles/landed_titles.txt]: Well, [Khalka Mongols](https://en.wikipedia.org/wiki/Khalkha_Mongols) do exist, but I don't know what this refers to in particular. Presumably it should be `khalan`, yes.

File: WTWSMS/decisions/WtWSMS_diploaction_decisions.txt [hunk 8]
```diff
@@ -96,4 +96,79 @@ targetted_decisions = {
...
+				NOT = { trait = incapable }
+				is_inaccessible_trigger = no
+			}
+		}
+		
+		effect = {
+			make_tributary = {
+				who = FROM
+				tributary_type = foederati
+			}
```
Comment (loup99) [inline WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: Is your intent to change this to event later on, or do you personally feel done with expanding this as it is?

Comment (IhateTrains) [reply WTWSMS/decisions/WtWSMS_diploaction_decisions.txt]: The negotiations have to be written, but the current decision should work well, I did some testing.

File: WTWSMS/common/landed_titles/landed_titles.txt (hunk 7 above)
Comment (IhateTrains) [reply WTWSMS/common/landed_titles/landed_titles.txt]: Done.
